src/Models/Nonhydrostatic3DModels/equations_thermodynamics.jl

```diff
@@ -129,3 +129,71 @@ end
     # direct stiffness summation
     Spaces.weighted_dss!(dρe_tot)
 end
+
+@inline function rhs_thermodynamics!(
+    dY,
+    Y,
+    Ya,
+    t,
+    p,
+    ::AdvectiveForm,
+    ::InternalEnergy,
+    params,
+    hyperdiffusivity,
+    flux_correction,
+    FT,
+)
+    # parameters
+    κ₄ = hyperdiffusivity
+
+    # unpack needed variables
+    dρe_int = dY.thermodynamics.ρe_int
+    ρ = Y.base.ρ
+    uh = Y.base.uh
+    w = Y.base.w
+    ρe_int = Y.thermodynamics.ρe_int
+
+    # operators /w boundary conditions
+    hdiv = Operators.Divergence()
+    hwdiv = Operators.Divergence()
+    hgrad = Operators.Gradient()
+    scalar_vgrad_c2f = Operators.GradientC2F(
+        bottom = Operators.SetGradient(Geometry.Covariant3Vector(FT(0))),
+        top = Operators.SetGradient(Geometry.Covariant3Vector(FT(0))),
+    )
+    vector_vdiv_f2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(Geometry.WVector(FT(0))),
+        top = Operators.SetValue(Geometry.WVector(FT(0))),
+    )
+    interp_c2f = Operators.InterpolateC2F(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+    interp_f2c = Operators.InterpolateF2C()
+    flux_correction_center = Operators.FluxCorrectionC2C(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+
+    # hyperdiffusion
+    χe = @. dρe_int = hwdiv(hgrad((ρe_int + p) / ρ))
+    Spaces.weighted_dss!(dρe_int)
+    @. dρe_int = -κ₄ * hwdiv(ρ * hgrad(χe))
+
+    # advection 
+    uvw = @. Geometry.Covariant123Vector(uh) +
+             Geometry.Covariant123Vector(interp_f2c(w))
+    # @. dρe_int -= hdiv(uvw * (ρe_int)) + p * hdiv(uvw)
+    # @. dρe_int -= vector_vdiv_f2c(w * interp_c2f(ρe_int)) + p * vector_vdiv_f2c(w)
+    # @. dρe_int -= vector_vdiv_f2c(interp_c2f(uh * (ρe_int))) + p * vector_vdiv_f2c(interp_c2f(uh))
```

Comment:
```suggestion
```

Suggest deleting these to avoid confusion (we can include the alternative form in the design-docs for reference).